client/network/sync/src/warp.rs

```diff
@@ -70,21 +80,54 @@ where
 	B: BlockT,
 	Client: HeaderBackend<B> + ProofProvider<B> + 'static,
 {
-	///  Create a new instance.
-	pub fn new(client: Arc<Client>, warp_sync_provider: Arc<dyn WarpSyncProvider<B>>) -> Self {
+	/// Create a new instance. When passing a warp sync provider we will be checking for proof and
+	/// authorities. Alternatively we can pass a target block when we want to skip downloading
+	/// proofs, in this case we will continue polling until the target block is known.
+	pub fn new(client: Arc<Client>, warp_sync_params: WarpSyncParams<B>) -> Self {
 		let last_hash = client.hash(Zero::zero()).unwrap().expect("Genesis header always exists");
-		let phase = Phase::WarpProof {
-			set_id: 0,
-			authorities: warp_sync_provider.current_authorities(),
-			last_hash,
+		match warp_sync_params {
+			WarpSyncParams::WithProvider(warp_sync_provider) => {
+				let phase = Phase::WarpProof {
+					set_id: 0,
+					authorities: warp_sync_provider.current_authorities(),
+					last_hash,
+					warp_sync_provider: warp_sync_provider.clone(),
+				};
+				Self { client, phase, total_proof_bytes: 0 }
+			},
+			WarpSyncParams::WaitForTarget(block) => Self {
+				client,
+				phase: Phase::PendingTargetBlock { target_block: block },
+				total_proof_bytes: 0,
+			},
+		}
+	}
+
+	/// Poll to make progress.
+	///
+	/// This only makes progress when `phase = Phase::PendingTargetBlock` and the pending block was
+	/// sent.
+	pub fn poll(&mut self, cx: &mut std::task::Context) {
+		let new_phase = if let Phase::PendingTargetBlock { target_block } = &mut self.phase {
+			match target_block.poll_unpin(cx) {
+				Poll::Ready(Ok(target_block)) => Phase::TargetBlock(target_block),
+				Poll::Ready(Err(e)) => {
+					error!(target: "sync", "Failed to get target block. Error: {:?}",e);
+					return
+				},
```

Comment:
When we reach this. This will be printed very often because polling will always return an error from that point. 

I know that I said that it is okay that we don't bring down the node, because with the current implementation this is a little bit hard (not the fault of this pr!). So, we should introduce some `bool` inside the `PendingTargetBlock` variant that is set to `true` when we have encountered the first error. Then we should also create some follow up issue to bring down the node when we can receiving the target block fails. This will require this pr: https://github.com/paritytech/substrate/pull/13219

Reply:
Sure but would it make sense to create a new pr to add the bool after https://github.com/paritytech/substrate/pull/13219 is merged, or would you prefer I do it as part of this pr?

Reply:
> So, we should introduce some `bool` inside the `PendingTargetBlock` variant that is set to `true` when we have encountered the first error.

This is the quick fix, so yes. The rest I said there is to remove the quick fix.

Reply:
changed the value of the target block to an optional value that is set to `None` when an error occurs